tests/components/vallox/test_config_flow.py

```diff
@@ -0,0 +1,242 @@
+    RESULT_TYPE_FORM,
+)
+
+
+async def test_form_no_input(hass: HomeAssistant) -> None:
+    """Test that the form is returned with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] is None
+
+
+async def test_form_create_entry(hass: HomeAssistant) -> None:
+    """Test that an entry is created with valid input."""
+    name = "Vallox 110 MV"
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: "1.2.3.4", CONF_NAME: name},
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"host": "unknown"}
+
+    with patch(
+        "vallox_websocket_api.Vallox.get_info",
+        return_value={"model": "Vallox 110 MV"},
+    ), patch(
+        "homeassistant.components.vallox.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"host": "1.2.3.4", "name": name},
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == name
+    assert result2["data"] == {"host": "1.2.3.4", "name": "Vallox 110 MV"}
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_host(hass: HomeAssistant) -> None:
+    """Test that invalid host error is handled."""
+    init = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        init["flow_id"],
+        {"host": "test.host&.com"},
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"host": "invalid_host"}
+
+
+async def test_form_vallox_api_exception_cannot_connect(hass: HomeAssistant) -> None:
+    """Test that cannot connect error is handled."""
+    init = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    with patch(
+        "vallox_websocket_api.Vallox.get_info",
+        side_effect=ValloxApiException,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            init["flow_id"],
+            {"host": "4.3.2.1"},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"host": "cannot_connect"}
+
+
+async def test_form_os_error_cannot_connect(hass: HomeAssistant) -> None:
+    """Test that cannot connect error is handled."""
+    init = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    with patch(
+        "vallox_websocket_api.Vallox.get_info",
```

Comment:
Please patch where the function is imported instead of globally

Reply:
I'm not sure I understand how to do that, would you pointing me in the right direction? It's definitely not the first time I'm writing tests, but it is the first time in Python 😁

Reply:
Done according to https://github.com/home-assistant/core/pull/62780#discussion_r775279467.